Mathlib/RingTheory/DedekindDomain/Ideal.lean

```diff
@@ -1538,4 +1539,40 @@ theorem multiplicity_normalizedFactorsEquivSpanNormalizedFactors_symm_eq_multipl
     multiplicity_normalizedFactorsEquivSpanNormalizedFactors_eq_multiplicity hr ha]
 #align multiplicity_normalized_factors_equiv_span_normalized_factors_symm_eq_multiplicity multiplicity_normalizedFactorsEquivSpanNormalizedFactors_symm_eq_multiplicity
 
+/-- The bijection `normalized_factors_equiv_span_normalized_factors.symm` between the set of prime
+    factors of the ideal `⟨r⟩` and the set of prime factors of `r` preserves `count` of the
+    corresponding multisets. See
+    `multiplicity_normalizedFactorsEquivSpanNormalizedFactors_eq_multiplicity` for the version
+    stated in terms of multiplicity. -/
+theorem count_normalizedFactorsSpan_eq_count {r X : R} (hr : r ≠ 0) (hX₁ : normUnit X = 1)
+    (hX : Prime X) : Multiset.count X (normalizedFactors r) =
+      Multiset.count (Ideal.span {X} : Ideal R) (normalizedFactors (Ideal.span {r})) := by
+    have := multiplicity_eq_multiplicity_span (R := R) (a := X) (b := r)
+    rw [multiplicity_eq_count_normalizedFactors (Prime.irreducible hX) hr,
+        multiplicity_eq_count_normalizedFactors] at this
+    · simp_all only [ne_eq, normalize_apply, normUnit_eq_one, Units.val_one, one_eq_top, mul_top,
+      mul_one, Nat.cast_inj]
+    · apply Prime.irreducible
+      dsimp only [Set.mem_setOf_eq, Set.coe_setOf, normalizedFactorsEquivSpanNormalizedFactors,
+        Equiv.ofBijective_apply]
+      simpa only [prime_span_singleton_iff]
+    · simp only [Submodule.zero_eq_bot, ne_eq, span_singleton_eq_bot, hr, not_false_eq_true]
+
+open Classical in
+/-- The number of times an ideal `I` occurs as normalized factor of another ideal `J` is stable
+  when regarding at these ideals as associated elements of the monoid of ideals.-/
+theorem count_normalizedFactors_eq_associates_count
+    (R : Type*) [CommRing R] [IsDomain R] [IsPrincipalIdealRing R]--cannot comment this!
```

Comment:
What do you mean by: `--cannot comment this!`

Reply:
Well, I know what you mean but I don't know why you felt necessary to add this comment

Reply:
I don't know what you mean. Are you thinking Lean should find the `DecidableEq` on `Associates` automatically from the existing context?

Reply:
Because all these variables are in scope, so I would expect that it is not necessary to call them again. In the previous theorem none of them was needed. Is it because of the `open Classical`?

Reply:
Why not add the requisite decidability instance to `Ideal.count_normalizedFactors_eq` and then to here?